src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/OffenderBuilder.kt

```diff
@@ -13,6 +13,7 @@ class OffenderBuilder(
   var genderCode: String = "M",
   var bookingBuilders: List<OffenderBookingBuilder> = mutableListOf(),
   val repository: Repository? = null,
+  val courseAllocationBuilderFactory: CourseAllocationBuilderFactory? = null,
```

Comment:
When everything is done every builder will have a reference to each of its child builder factories. This is currently only nullable because we're in a halfway house and there is other usage of the `OffenderBuilder` outside of the new `TestData` DSL which need to carry on working.